arm/Microsoft.ServiceFabric/clusters/deploy.bicep

```diff
@@ -0,0 +1,398 @@
+  storageAccountName: !empty(diagnosticsStorageAccountConfig) ? diagnosticsStorageAccountConfig.storageAccountName : (!empty(supportLogStorageAccount) ? supportLogStorageAccount.name : null)
+  tableEndpoint: !empty(diagnosticsStorageAccountConfig) ? diagnosticsStorageAccountConfig.tableEndpoint : (!empty(supportLogStorageAccount) ? supportLogStorageAccount.properties.primaryEndpoints.table : null)
+}
+
+// Service Fabric cluster resource
+resource serviceFabricCluster 'Microsoft.ServiceFabric/clusters@2021-06-01' = {
+  name: serviceFabricClusterName
+  location: location
+  tags: tags
+  properties: {
+    addOnFeatures: addOnFeatures
+    applicationTypeVersionsCleanupPolicy: {
+      maxUnusedVersionsToKeep: maxUnusedVersionsToKeep
+    }
+    azureActiveDirectory: !empty(azureActiveDirectory) ? azureActiveDirectory_var : null
+    certificate: !empty(certificate) ? certificate_var : null
+    certificateCommonNames: !empty(certificateCommonNames) ? certificateCommonNames_var : null
+    clientCertificateCommonNames: !empty(clientCertificateCommonNames) ? clientCertificateCommonNames_var : null
+    clientCertificateThumbprints: !empty(clientCertificateThumbprints) ? clientCertificateThumbprints_var : null
+    clusterCodeVersion: !empty(clusterCodeVersion) ? clusterCodeVersion : null
+    diagnosticsStorageAccountConfig: !empty(diagnosticsStorageAccountConfig) ? diagnosticsStorageAccountConfig_var : null
+    eventStoreServiceEnabled: eventStoreServiceEnabled
+    fabricSettings: !empty(fabricSettings) ? fabricSettings_var : null
+    infrastructureServiceManager: infrastructureServiceManager
+    managementEndpoint: '${!empty(managementEndpoint) ? managementEndpoint : null}'
+    nodeTypes: nodeTypes_var
+    notifications: !empty(notifications) ? notifications_var : null
+    reliabilityLevel: reliabilityLevel
+    reverseProxyCertificate: !empty(reverseProxyCertificate) ? reverseProxyCertificate_var : null
+    reverseProxyCertificateCommonNames: !empty(reverseProxyCertificateCommonNames) ? reverseProxyCertificateCommonNames_var : null
+    sfZonalUpgradeMode: !empty(sfZonalUpgradeMode) ? sfZonalUpgradeMode : null
+    upgradeDescription: !empty(upgradeDescription) ? upgradeDescription_var : null
+    upgradeMode: !empty(upgradeMode) ? upgradeMode : null
+    upgradePauseEndTimestampUtc: !empty(upgradePauseEndTimestampUtc) ? upgradePauseEndTimestampUtc : null
+    upgradePauseStartTimestampUtc: !empty(upgradePauseStartTimestampUtc) ? upgradePauseStartTimestampUtc : null
+    upgradeWave: !empty(upgradeWave) ? upgradeWave : null
+    vmImage: !empty(vmImage) ? vmImage : null
+    vmssZonalUpgradeMode: !empty(vmssZonalUpgradeMode) ? vmssZonalUpgradeMode : null
+    waveUpgradePaused: waveUpgradePaused
+  }
+}
+
+// Other Service fabric post requisites
+module serviceFabricPrerequisites '.bicep/nested_postrequisites.bicep' = {
+  name: '${serviceFabricCluster.name}-Postreqs'
```

Comment:
Not sure I understand what is happening here

Reply:
Mistakenly got propagated after testing, removed now.